build-logic/packaging/src/main/kotlin/gradlebuild.shaded-jar.gradle.kts

```diff
@@ -47,6 +47,14 @@ addInstallShadedJarTask(shadedJarTask)
 addShadedJarVariant(shadedJarTask)
 configureShadedSourcesJarVariant()
 
+plugins.withId("gradlebuild.publish-public-libraries") {
+    gradleModule {
+        // Since all of our dependencies are shaded, we don't care if they are published or not.
+        // Hackily declare this project as non-published to skip the verification.
+        published = false
```

Comment:
💅 IIUC this hack would fail if, for example, the `tooling-api` were to be depended upon by a published other project?

Reply:
Yes it would. To make this less of a hack we would want to instead add a flag that disables adding this attribute to the resolvable configurations, so that we don't request that our dependencies are published. 

However, hopefully we can integrate the verification into gradle, where this would not be a problem anymore (we would check during publication instead, which takes into account the shaded dependencies)